File: BackEnd/src/main/java/com/tienda_ropa/ecommerce/model/Promocion.java
package com.tienda_ropa.ecommerce.model;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;

import java.time.*;
import java.util.*;

@Entity
@Table(name = "promocion")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Promocion extends Master {

    @Column(name = "fecha_desde")
    private LocalDateTime fechaDesde;

    @Column(name = "fecha_hasta")
    private LocalDateTime fechaHasta;

    @Column(name = "hora_desde")
    private LocalTime horaDesde;

    @Column(name = "hora_hasta")
    private LocalTime horaHasta;

    @Column(name = "tipo_promocion")
    private String tipoPromocion;

    private String descripcion;

    @OneToMany(mappedBy = "promocion", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference
    private Set<ImagenPromocion> imagenes = new HashSet<>();

    @OneToMany(mappedBy = "promocion", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference
    private Set<PromocionDetalle> detalles = new HashSet<>();
}

File: FrontEnd/src/models/Promocion.ts
import type ImagenPromocion from "./ImagenPromocion";
import PromocionDetalle from "./PromocionDetalle";

export default class Promocion {
    id?: number;
    denominacion: string = "";
    fechaDesde: Date = new Date();
    fechaHasta: Date = new Date();
    horaDesde: Date = new Date();
    horaHasta: Date = new Date();
    descripcionDescuento: string = "";
    precioPromocional: number = 0;
    tipo?: String;
    imagenes?: ImagenPromocion[];
    detalle?: PromocionDetalle[];    activa: boolean = false;
    eliminado!: boolean;
}